kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActionsIterator.java

```diff
@@ -195,7 +209,9 @@ private CloseableIterator<ColumnarBatch> getActionsIterFromSinglePartOrV2Checkpo
       checkpointPredicateIncludingSidecars = checkpointPredicate;
     }
     final CloseableIterator<ColumnarBatch> topLevelIter;
-    StructType finalModifiedReadSchema = modifiedReadSchema;
+    StructType finalJsonReadSchema = modifiedJsonReadSchema;
```

Comment:
Ah, sorry -- I am noticing a small disconnect here.

So, last PR review I did suggest to use `jsonReadSchema` and `parquetReadSchema`.

I'm realizing this is a bit incorrect. We do want `deltaReadSchema` (for delta `.json` files) and `checkpointReadSchema`. Checkpoints are usually parquet, but they CAN be json in the case of V2 checkpoints. That's why `jsonReadSchema` and `parquetReadSchema` are a bit wrong.

Sorry for realizing this only now :( 

Can you please

(1) use `deltaReadSchema` with a comment what that really means and (2) `checkpointReadSchema`, also with a good comment explaining how this can be both parquet and json files

(2) inside of `getActionsIterFromSinglePartOrV2Checkpoint` -- you only need to use the `checkpointReadSchema`

Thanks!

Reply:
Done.